consensus/safety-rules/src/safety_rules.rs

```diff
@@ -217,6 +199,26 @@ impl SafetyRules {
         Err(Error::IncorrectLastVotedRound(round, last_voted_round))
     }
 
+    /// Second voting rule
+    fn verify_voting_rule(
+        &self,
+        block: &Block,
+        maybe_tc: Option<&TimeoutCertificate>,
+    ) -> Result<(), Error> {
+        let block_round = block.block_data().round();
+        let parent_round = block.quorum_cert().certified_block().round();
+        if block_round == next_round(parent_round)? {
+            return Ok(());
+        }
+        if let Some(tc) = maybe_tc {
+            if block_round == next_round(tc.round())? && parent_round >= tc.hqc_round() {
```

Comment:
let's also check block_round > parent_round here, unless it's actually checked elsewhere (not done here now). This seems to simplify the 2-chain agreement proof (avoid dealing with bizarre scenarios in which some validators make faster progress and send a proposal extending the future). Since it's a really straightforward & intuitive check, I think adding it to match the proved version is better than complicating the proof.